app/controllers/user_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Chitter < Sinatra::Base
+  get '/sign_up' do
+    erb :'user/new'
+  end
+
+  post '/register_user' do
+    user = create_user(params)
+    create_session_data(user)
+    redirect '/sign_up' unless user.id
```

Comment:
I think using 'unless' here makes it feel like user.id is the exception case, rather than expected case. I'd rather see redirect `"user/..." unless user.id == nil `etc, it makes it clearer that 'normally we do this, unless it has failed, then we do this'